### PR TITLE
Roll the dice example for elixir/phoenix

### PR DESCRIPTION
**Pull request by marcdel**
heavily cribbed from https://github.com/open-telemetry/opentelemetry.io/pull/2624

- [ ] squash these commits

---

Preview: https://deploy-preview-2685--opentelemetry.netlify.app/docs/instrumentation/erlang/getting-started/

### CODE REVIEW

File: content/en/docs/instrumentation/erlang/getting-started/rebar.md
Comment (chalin) [inline content/en/docs/instrumentation/erlang/getting-started/rebar.md:3]: No need for an alias:

```suggestion
```

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (chalin) [inline content/en/docs/instrumentation/erlang/getting-started/phoenix.md:3]: No need for an alias:

```suggestion
```

File: content/en/docs/instrumentation/erlang/getting-started/_index.md
Comment (chalin) [inline content/en/docs/instrumentation/erlang/getting-started/_index.md:3]: No need for an alias:

```suggestion
```

**Review by chalin — COMMENTED** [review]
Very preliminary feedback: drop the `aliases` entries, they aren't needed.

File: content/en/docs/instrumentation/erlang/getting-started/rebar.md
Comment (marcdel) [reply content/en/docs/instrumentation/erlang/getting-started/rebar.md]: thanks! totally just cargo-culted these from the other doc

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (tsloughter) [inline content/en/docs/instrumentation/erlang/getting-started/phoenix.md:28]: I think instead of `your_app_name` this should just use the example app name that'll go in the -contrib repo examples.

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (tsloughter) [inline content/en/docs/instrumentation/erlang/getting-started/phoenix.md:48]: Ecto should be left out from the example since it isn't used. We want this to start without a database. But since it is likely any user will have Ecto it should definitely include a link and mention of `opentelemetry_ecto`.

Should also add a link to the demo, https://github.com/open-telemetry/opentelemetry-demo/tree/main/src/featureflagservice, and examples dir of contrib, https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/examples/, saying something like, 'for more complete Phoenix/Ecto projects see these'.

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (tsloughter) [inline content/en/docs/instrumentation/erlang/getting-started/phoenix.md:204]: Should give some details about how this is showing the span name and attributes for 2 spans.

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (tsloughter) [inline content/en/docs/instrumentation/erlang/getting-started/phoenix.md:154]:
```suggestion
These are the raw Erlang records that will get serialized and sent when you configure the
```

**Comment by tsloughter** [comment]
Had a few comments but overall this looks great.

**Comment by oliverswitzer** [comment]
👏👏 thanks for doing this @marcdel!

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (marcdel) [reply content/en/docs/instrumentation/erlang/getting-started/phoenix.md:28]: Question about this: there's a [basic_phoenix_ecto](https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/examples/basic_phoenix_ecto), I assume this would be a separate `dice` (`dice_game`?) example in addition to that one?

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (marcdel) [reply content/en/docs/instrumentation/erlang/getting-started/phoenix.md:48]: Should the example be explicitly created with `--no-ecto`?

Comment (tsloughter) [reply content/en/docs/instrumentation/erlang/getting-started/phoenix.md:48]: Yea.

File: content/en/docs/instrumentation/erlang/getting-started/phoenix.md
Comment (tsloughter) [reply content/en/docs/instrumentation/erlang/getting-started/phoenix.md:28]: Yup, a separate dice game example.

**Comment by tsloughter** [comment]
2The reason is in part from #587 where there needs to be a single attribute to expose the schema url for the instrumentation library. So need bbb

**Comment by svrnm** [comment]
@marcdel, @tsloughter not knowing enough about erlang/elixir, so I leave that to you. I have one question/concern I'd like to clarify: One of the goals we try to accomplish with standardizing on one application for "Getting Started" is to get consistency across the docs. While this PR accomplishes that partially I still see some inconsistencies I'd like to see if we can resolve them:

Screenshot 2023-05-15 at 10 41 16

There are now 2 kinds of getting started, which is confusing. Would it be possible to make the "Phoenix" one with the roll the dice app the REAL getting started page and take whatever is important from the existing Getting Started and merge it and put everything else into pages where it fits (e.g. exporters have their own page across the languages)

Also, if I look at the Phoenix app there is some differences to other languages. Would it be possible to structure the page in a similar way (e.g. start with a ready to use roll the dice app and instrument from there)

**Note**: We can accomplish some/all of that in follow-up PRs but I wanted to clarify that.

 cc @cartermp @chalin @austinlparker

**Comment by tsloughter** [comment]
@svrnm ok, yea, I agree about starting from a working roll dice app and then instrument that. The README.md of the example can have any setup instructions about how the project was created. What do you think @marcdel ?

I think we can also move stuff into a single `Getting Started` and add some stuff to the index page about adding deps. The main use is certainly going to be adding to existing projects, and there are plenty of other guides on creating Erlang and Elixir projects.

**Comment by svrnm** [comment]
@marcdel this is really great work so I am eager to merge this in, please let us know if and where you might need assistance, if any of our comments or change requests are unclear/a too big ask, we can clarify and help with the heavy lifting.

**Comment by marcdel** [comment]
> **Note**: We can accomplish some/all of that in follow-up PRs but I wanted to clarify that.

> There are now 2 kinds of getting started, which is confusing. Would it be possible to make the "Phoenix" one with the roll the dice app the REAL getting started page and take whatever is important from the existing Getting Started and merge it and put everything else into pages where it fits (e.g. exporters have their own page across the languages)

@svrnm Does it make sense to combine these two pages and get this merged in, then address those other bits? I'd rather add pages than remove ones someone might already be linking to.

FWIW I was following the [javascript](https://deploy-preview-2685--opentelemetry.netlify.app/docs/instrumentation/js/getting-started/) one which has node as a child of the getting started page. I don't have any strong feelings on organization so I'm happy to do whatever!

**Comment by marcdel** [comment]
Okie dokie, I've moved the Phoenix stuff into the main Getting Started page and shuffled some things around such that it hopefully makes sense. I also moved the setup bits over to the readme of https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/173, and edited the surrounding bits to assume all the setup is already done.

**Comment by tsloughter** [comment]
Thanks. I think this will work for now and I'll merge together the Phoenix and Erlang getting started once I write the Erlang part. So it won't have the section about creating a new project anymore once I do that.

**Review by cartermp — APPROVED** [review]
Thanks @marcdel, this is great. I'll merge this assuming nobody else has any objections in a day or so.

**Pull request merged**
